client/lxd_storage_volumes.go

```diff
@@ -362,9 +362,26 @@ func (r *ProtocolLXD) MigrateStoragePoolVolume(pool string, volume api.StorageVo
 		return nil, fmt.Errorf("Can't ask for a rename through MigrateStoragePoolVolume")
 	}
 
+	var req any
+	var path string
+
+	srcVolParentName, srcVolSnapName, srcIsSnapshot := api.GetParentAndSnapshotName(volume.Name)
+	if srcIsSnapshot {
+		// Set the actual name of the snapshot without delimiter.
+		req = api.StorageVolumeSnapshotPost{
+			Name:      srcVolSnapName,
+			Migration: volume.Migration,
+			Target:    volume.Target,
+		}
+
+		path = fmt.Sprintf("/storage-pools/%s/volumes/custom/%s/snapshots/%s", url.PathEscape(pool), srcVolParentName, srcVolSnapName)
```

Comment:
We should use url.PathEscape() on all path parameters.

Reply:
I switched to using `api.NewURL().Path().String()` to remove the burden of having to call `url.PathEscape()` countless times.

Reply:
Yes that is preferred way of doing it indeed.